bpf/aya-bpf/src/args.rs

```diff
@@ -56,6 +56,30 @@ unsafe_impl_from_btf_argument!(i64);
 unsafe_impl_from_btf_argument!(usize);
 unsafe_impl_from_btf_argument!(isize);
 
+pub struct PtRegs {
+    regs: *mut pt_regs,
+}
+
+/// PtRegs is a portable wrapper around
+/// pt_regs and user_pt_regs
+impl PtRegs {
+    pub fn new(ctx: *mut c_void) -> Self {
+        PtRegs {
```

Comment:
Not sure if pt_regs will always the first element of a context, better leave the cast to the caller.

Reply:
That's correct, but `new` isn't only callable using a `ctx` - any `*mut c_void` will do - have renamed the arg to make that clearer.

example code from a kprobe on  `__x64_sys_execve`, where `ctx.arg(0)` is a `pt_regs`
```rust
let pt_regs = PtRegs::new(ctx.arg(0).unwrap());
```